### PR TITLE
Include registrations in dashboard search

### PR DESCRIPTION
**Pull request by irisfaraway**
https://eaflood.atlassian.net/browse/RUBY-704

The back office dashboard search was originally designed to only look at renewals - records in the TransientRegistrations table. We want to look at the Registrations table as well so we can see completed registrations and new applications in progress.

This will require updating our search service, as well as changing the design of the dashboard to be able to handle both types of object and show the correct information and actions for them.

### CODE REVIEW

**Comment by irisfaraway** [comment]
Relies on https://github.com/DEFRA/waste-carriers-engine/pull/511

**Comment by irisfaraway** [comment]
This was starting to get hefty so I've rebased it into several PRs. This does the work of updating the search so that we get the registrations into the results, although they're not displayed the way that we ultimately want them to be.

Once this is merged, we can look at:

- #384 
- #385 
- #386

File: app/views/dashboards/index.html.erb [hunk 1]
```diff
@@ -79,6 +79,8 @@
                 <% end %>
               </td>
               <td>
+                <% if result.is_a?(WasteCarriersEngine::TransientRegistration) %>
+
                 <% if result_type(result).present? %>
```
Comment (cintamani) [inline app/views/dashboards/index.html.erb]: Can we fix indentation? :D

Comment (irisfaraway) [reply app/views/dashboards/index.html.erb]: I think this gets removed by another PR anyway as it was a temporary fix :)

**Pull request merged**